Softeq.XToolkit.Bindings/BindingExtensions.cs

```diff
@@ -525,93 +479,93 @@ public static void SetCommand(this object element, string eventName, Action acti
-        ///     overloads that uses a Binding as CommandParameter.
+        /// <param name="commandParameterBinding">
+        ///     A <see cref="Binding{TSource, TTarget}" /> instance subscribed to the CommandParameter
+        ///     that will passed to the <see cref="ICommand"/>.
+        ///     Depending on the <see cref="Binding"/>, the CommandParameter will be observed and changes
+        ///     will be passed to the command, for example to update the CanExecute.
         /// </param>
         public static void SetCommand<T, TEventArgs>(
             this object element,
             string eventName,
             ICommand command,
-            T commandParameter)
+            Binding commandParameterBinding)
         {
             var t = element.GetType();
             var e = t.GetEventInfoForControl(eventName);
 
-            EventHandler<TEventArgs> handler = (s, args) => command.Execute(commandParameter);
+            var castedBinding = (Binding<T, T>) commandParameterBinding;
 
-            e.AddEventHandler(
-                element,
-                handler);
+            var handler = _bindingFactory.GetCommandHandler<T, TEventArgs>(e, eventName, t, command, castedBinding);
 
-            var enabledProperty = t.GetRuntimeProperty("Enabled");
-            if (enabledProperty != null)
-            {
-                enabledProperty.SetValue(element, command.CanExecute(commandParameter));
+            e.AddEventHandler(element, handler);
 
-                command.CanExecuteChanged += (s, args) => enabledProperty.SetValue(
-                    element,
-                    command.CanExecute(commandParameter));
+            if (commandParameterBinding == null)
+            {
+                return;
             }
-        }
 
-        public static void SetCommand<T>(
-            this object element,
-            ICommand command,
-            Binding commandParameterBinding)
-        {
-            SetCommand(element, string.Empty, command, commandParameterBinding);
+            HandleEnabledProperty(element, t, command, castedBinding);
         }
 
-        public static void SetCommand<T, TEventArgs>(
+        /// <summary>
+        ///     Sets a <see cref="ICommand"/> to an object and actuates the command when a specific event is raised.
+        ///     This method can only be used when the event uses a standard <see cref="EventHandler"/>.
+        /// </summary>
+        /// <param name="element">The element to which the command is added.</param>
+        /// <param name="eventName">The name of the event that will be subscribed to to actuate the command.</param>
+        /// <param name="command">The command that must be added to the element.</param>
+        /// <returns><see cref="IDisposable"/> instance for manual unset/unsubscribe of command.</returns>
+        public static IDisposable SetCommandWithDisposing(
```

Comment:
If I'm not mistaken, with current implementation there's only one way to set command with disposing and a lot of overloaded methods for setting command without disposing. So it might be as convenient to use disposing variant..
Making the same set of overloaded methods for disposing doesn't sound like much fun, but maybe we could make all of the methods returning disposable by default? It won't affect usage of SetCommand() methods - returned value will be just ignored, but it will improve flexibility of API when disposing is needed

Reply:
Yes, I agree with you)
I wanted to add only one method firstly. After PR review or additional discussion, I planned to collect opinions of apply `Disposable` for all methods.

Reply:
If we gonna pick only one method to test Disposable implementations - may I suggest to chose signature
`IDisposable SetCommandWithDisposing(this object element, ICommand command)`
We use it in cells, so we would be able to test it right away

Reply:
```cs
IDisposable SetCommandWithDisposing(this object element, ICommand command)
```
will be the same as this:
```cs
element.SetCommandWithDisposing(string.Empty, ICommand command)
```

Reply:
I understand. It's just that if we agree to use Disposing implementation by default - we would then  go through the code once again and remove redundant parameter